### PR TITLE
ref: `remote modify` `gdrive_acknowledge_abuse` and `profile`

### PR DESCRIPTION
**Pull request by efiop**
Per https://github.com/iterative/dvc/pull/8040
and https://github.com/iterative/dvc/pull/8060

### CODE REVIEW

**Comment by efiop** [comment]
Ah, need to add `profile` along the way. Will get back to this shortly.

**Review by shcheklein — APPROVED** [review]

**Review by shcheklein — APPROVED** [review]

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:608]:
```suggestion
- `profile` - name for the credentials cache (part of the default
  `gdrive_user_credentials_file` path). By default, credentials will be cached
  per `gdrive_client_id `, but if you have multiple system users, you can set a
  custom ‘profile’ name to avoid using someone else’s credentials accidentally.
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:632]:
```suggestion
- `gdrive_acknowledge_abuse` - acknowledge the risk of downloading potentially
  [abusive](https://support.google.com/docs/answer/148505). Files identified
  as such (malware, personal info., etc.) can only be downloaded by their
  owner.
```

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:632]: > Consider using this option if you see a
  `This file has been identified as malware or spam and cannot be downloaded`
  error.

We should add a [troubleshooting](https://dvc.org/doc/user-guide/troubleshooting) entry instead, I think.

File: content/docs/contributing/core.md
Comment (jorgeorpinel) [inline content/docs/contributing/core.md:233]:
```suggestion
DVC will automatically cache the Google Drive access token obtained in a global
location (e.g. `~/.cache/pydrive2fs` on Linux, see [details]) to avoid manual
logins after this.

[details]: https://dvc.org/doc/command-reference/remote/modify#google-drive
```

Comment (jorgeorpinel) [reply content/docs/contributing/core.md:233]: Would be ideal to move the `[details]` ref link to the bottom of the section.

File: content/docs/user-guide/privacy.md
Comment (jorgeorpinel) [inline content/docs/user-guide/privacy.md:72]:
```suggestion
By default, OAuth tokens are cached in a global location (e.g.
`~/.cache/pydrive2fs` on Linux, see [details]).

[details]: https://dvc.org/doc/command-reference/remote/modify#google-drive
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:185]:
```suggestion
will be cached (see `gdrive_user_credentials_file` in the [GDrive remote
params]) and used automatically next time DVC needs them.

[gdrive remote params]:
  https://dvc.org/doc/command-reference/remote/modify#google-drive
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:193]:
```suggestion
cached credentials. This can be overridden with
`dvc remote modify`:
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:205]:
```suggestion
troubleshooting misc. token errors, you can remove the user credentials file
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:207]:
```suggestion
and authorize again. Alternatively you can use a custom `profile`-based
credential cache (instead of the default, `gdrive_client_id`-based one):
```

**Review by jorgeorpinel — COMMENTED** [review]
Bunch of edits -- committing...

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:588]:
```suggestion
- `gdrive_user_credentials_file` - global path where DVC stores OAuth credentials to
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:608]:
```suggestion
- `profile` - name for the GDrive credentials cache directory (part of the
  global `gdrive_user_credentials_file` path). By default, credentials are cached
  per `gdrive_client_id `, but you can set a custom ‘profile’ instead, for example
  to avoid using the wrong user's credentials accidentally.
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:630]:
```suggestion
  [abusive](https://support.google.com/docs/answer/148505) data. Files
  identified as such (malware, personal info., etc.) can only be downloaded
  by their owner (with this param enabled).
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:197]:
```suggestion
cached credentials. This can be overridden with `dvc remote modify`:
```

**Review by jorgeorpinel — COMMENTED** [review]
More edits... (committing)

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:589]: Format
```suggestion
- `gdrive_user_credentials_file` - global path where DVC stores OAuth
  credentials to access Google Drive, by default
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:608]: Typo
```suggestion
  per `gdrive_client_id `, but you can set a custom profile name instead, for
  example to avoid using the wrong user's credentials accidentally.
```

**Review by jorgeorpinel — CHANGES_REQUESTED** [review]
Please merge https://github.com/iterative/dvc.org/pull/3810 instead if you agree with my changes here, @efiop. Thanks

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:606]: > "part of the default gdrive_user_credentials_file path" - where does it come from? why do you think it's part of that path?

@shcheklein so that's not the case? What I understood from this change is that it's `$CACHE_HOME/pydrive2fs/{gdrive_client_id}/default.json` by default, but if you set a `profile` then it becomes `$CACHE_HOME/pydrive2fs/{profile}/default.json`? Cc @efiop

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:628]: > potentially abusive ... ? (files, content, something is needed here)

@shcheklein It's "data" here. Not sure why #3810 doesn't have that word... ⏳

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: > let's keep this part: "Consider using this option if you see ..." it's important for search, etc

@shcheklein my suggestion was to make a troubleshooting entry instead ☝🏼

WDYT @efiop ? I could also keep it for now and make an issue to move it to Troubleshooting later, lmk.

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:606]: Yes, except small correction - `$CACHE_HOME/pydrive2fs/{profile}/default.json` should be `$CACHE_HOME/pydrive2fs/{client_id}/profile.json`

Anyways, the question is why specifically mentioning this parameter - `gdrive_user_credentials_file` ? `gdrive_user_credentials_file` redefined it completely as far as I understand (`profile` then is not taken into account). cc @efiop could you confirm?

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:628]: why not files?

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md]: I think it's fine to keep the message here for now. It'll be searchable that's the most important thing.

File: content/docs/command-reference/remote/modify.md
Comment (efiop) [reply content/docs/command-reference/remote/modify.md:606]: I confirm 🙂

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:591]: Format
```suggestion
  `$CACHE_HOME/pydrive2fs/{gdrive_client_id}/default.json` (see also `profile`
  below), where the`CACHE_HOME` location per platform is:
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:630]: Format
```suggestion
  identified as such (malware, personal info., etc.) can only be downloaded by
  their owner (with this param enabled).
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:210]: Format
```suggestion
troubleshooting misc. token errors, you can remove the user credentials file and
authorize again. Alternatively you can use a custom `profile`-based credential
cache (instead of the default, `gdrive_client_id`-based one):
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:608]: `profile` update per https://github.com/iterative/dvc.org/pull/3795#discussion_r932519820
```suggestion
- `profile` - name for the GDrive credentials file (see also
  `gdrive_user_credentials_file` above). By default, credentials are cached per
  client ID but you can set a custom profile name instead, for example to avoid
  using the wrong user's credentials accidentally.
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:606]: > why specifically mentioning this parameter - gdrive_user_credentials_file

Maybe I'm still confused. `profile` changes the default `gdrive_user_credentials_file`, right? Specifically the last part (file name). If so, there's a direct relationship.

**Review by jorgeorpinel — COMMENTED** [review]
Committing a few more updates:

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:606]: > why specifically mentioning this parameter - gdrive_user_credentials_file

Maybe I'm still confused. `profile` changes the default `gdrive_user_credentials_file`, right? Specifically the last part (file name). If so, there's a direct relationship.

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:628]: > why not files?

Because the next sentence starts with "Files".

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:628]: But OK, we can do this:
```suggestion
- `gdrive_acknowledge_abuse` - acknowledge the risk of downloading potentially
  [abusive](https://support.google.com/docs/answer/148505) files. Anything
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:608]: > small correction - should be `$CACHE_HOME/pydrive2fs/{client_id}/profile.json`

I'm still confused here. Why are we saying this value changes the caching by client ID if the the `{client_id}` part of the default `gdrive_user_credentials_file` is not changed by this?

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: > gdrive_user_credentials_file redefined it completely

What happens if you set both `profile` and `gdrive_user_credentials_file`? Which one wins?

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:608]: okay, `gdrive_user_credentials_file` defines actually full path

`profile` defines literally file name. I would move `see also
  `gdrive_user_credentials_file` above` to the end of the paragraph. Would give an example `$CACHE_HOME/pydrive2fs/{client_id}/{pfofile}.json` and be more explicit that if you want to completely change its location use that variable.

Does it answer your questions, @jorgeorpinel ?

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md]: `profile` is ignored in this case (if it's not it's a bug). See the comment here https://github.com/iterative/dvc.org/pull/3795#discussion_r932667184

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: More or less. I think I got it now (updated both guide and ref), PTAL...

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [inline content/docs/user-guide/setup-google-drive-remote.md:214]: Instead of giving an example where the custom credentials file location is inside the repo and then having to warn about not putting it in the repo, should the example be in the home directory? We can still warn against putting it in the repo though.

Comment (shcheklein) [reply content/docs/user-guide/setup-google-drive-remote.md:214]: yes, it makes sense to me

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:166]: let's keep it myprofile, it's not user (otherwise we would call it user)

Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:166]: also, can we please avoid touching files that are not related 🙏 :)

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:593]: same, should be `myprofile`

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:589]: hmm, I think the previous version was a bit better, more explicit ... how was it phrased? it not about passing wrong credentials only, it's about also rewriting wrong credentials

we are overcomplicating this, Jorge ...

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:598]: default is `default.json`, unless profile is specified

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:183]: let's put a proper example and see also ...

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:194]: they won't, only if it's only the same client also

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:199]: myprofile

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (shcheklein) [inline content/docs/user-guide/setup-google-drive-remote.md:223]: why touching this? is it related?

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: I would argue that it's related (for consistency) since it's in the same page. Changed to `myprofile` throughout this page and in the GDrive guide now.

> These small things I'll never get be able to get to again -- it's now or never... We keep having this misunderstanding 😢

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:589]: > how was it phrased?

"if you are using multiple users you might want to
  use ‘profile’ to avoid accidentally using someone else’s cached credentials"

- [original commit](https://github.com/iterative/dvc.org/pull/3795/commits/57dc421526c6fd650aacac5d3372e0b3dd523ae1#diff-a99e08a2de18082aa1adc45807e98187fe62f245a715f12cd08ec703823c9273R606-R608)

> it's about also rewriting...

Right, also only [if it's the same client](https://github.com/iterative/dvc.org/pull/3795#pullrequestreview-1055034448) I guess. Fixed.

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [reply content/docs/user-guide/setup-google-drive-remote.md]: > should the (`profile`) example be in the home directory 

Done.

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:598]: Then we have to elaborate on what part of the path `profile` changes:

```suggestion
  `$CACHE_HOME/pydrive2fs/{gdrive_client_id}/default.json` unless
  `profile` is specified (which changes the file name), where
```

Feel free to commit this, but I think the current explanation is simpler.

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [reply content/docs/user-guide/setup-google-drive-remote.md:183]: Not exactly sure what you mean. Proper example = `~/Library/Caches/...`? If so please commit:

```suggestion
globally, for example in
`~/Library/Caches/pydrive2fs/{gdrive_client_id}/default.json` for Linux
([see `gdrive_user_credentials_file`]), and used automatically next time
```

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [reply content/docs/user-guide/setup-google-drive-remote.md]: I guess not. For a moment I thought this PR also introduced `gdrive_user_credentials_file` but no. Reverted.

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (efiop) [reply content/docs/user-guide/setup-google-drive-remote.md:183]: @jorgeorpinel That's incorrect for linux. It is more like "~/.cache/pydrife2fs/{gdrive_client_id}/default.json", as indicated by the table
Screenshot 2022-07-29 at 20 53 42

File: content/docs/command-reference/remote/modify.md
Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:598]: let's do this and just add (see also `profile` ..) - that should be enough.

**Comment by shcheklein** [comment]
@jorgeorpinel hey, is it done? are we waiting for one more review?

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: Done.

File: content/docs/user-guide/setup-google-drive-remote.md
Comment (jorgeorpinel) [reply content/docs/user-guide/setup-google-drive-remote.md:183]: Right, I meant macOS. Changed. Committing...

**Comment by jorgeorpinel** [comment]
Looks like everything is addressed @shcheklein . Should we merge?

**Review by shcheklein — APPROVED** [review]

**Pull request merged**
